instrumentation/http_instrumentation.go

```diff
@@ -91,10 +102,67 @@ func (m instrumentedHandlerFactory) NewHandler(extraLabels prometheus.Labels, ne
 }
 
 // NewInstrumentedHandlerFactory creates a new instrumentedHandlerFactory.
-func NewInstrumentedHandlerFactory(req *prometheus.Registry, hardcodedLabels []string) instrumentedHandlerFactory {
+func NewInstrumentedHandlerFactory(req *prometheus.Registry) instrumentedHandlerFactory {
 	return instrumentedHandlerFactory{
-		metricsCollector: newHTTPMetricsCollector(req, hardcodedLabels),
+		metricsCollector: newHTTPMetricsCollector(req),
+	}
+}
+
+func httpHandlerLabels(r *http.Request) prometheus.Labels {
```

Comment:
Ok, so this seems very rigid and hardcoded. Why not keep it injectable as @PhilipGough mentioned and did previously here? 
> something to keep in mind is that moving this from being injectable to being hardcoded here makes the package unusable with any other routing framework

Is there some bug/issue you noticed with keeping it injectable @douglascamata?

Reply:
There's no reason to keep the added complexity if in reality we will not migrate to any different routing framework any time soon. We need something simple that works with what we have. Besides, how is this more rigid and hardcoded than the previous solution?

Reply:
In the end, the handlers themselves are using `Chi`, so they pretty much have a hard dependency on the exact router system we are using.